components/brave_wallet/browser/brave_wallet_utils.cc

```diff
@@ -1507,6 +1562,24 @@ mojom::CoinType GetCoinForKeyring(const std::string& keyring_id) {
   return mojom::CoinType::ETH;
 }
 
+absl::optional<std::string> CoinTypeToKeyringId(
+    mojom::CoinType coin_type,
+    const absl::optional<std::string>& chain_id) {
+  switch (coin_type) {
+    case mojom::CoinType::ETH:
+      return mojom::kDefaultKeyringId;
+    case mojom::CoinType::SOL:
+      return mojom::kSolanaKeyringId;
+    case mojom::CoinType::FIL:
+      if (!chain_id.has_value()) {
+        return mojom::kFilecoinKeyringId;
+      }
+      return GetFilecoinKeyringId(*chain_id);
+    default:
+      return absl::nullopt;
```

Comment:
NOTREACHED() here
and a tiny test as it is a public function.

Reply:
fixed and squashed.